_stbt/core.py

```diff
@@ -1907,7 +1906,6 @@ def create_source_pipeline(self):
             # Hauppauge HDPVR capture device.
             source_queue = self.source_pipeline.get_by_name(
                 "_stbt_user_data_queue")
-            self.start_timestamp = None
```

Comment:
`start_timestamp` was changed from a stack variable to a member in cf57a4c (First `wait_for_match` after restarting pipeline obeys `timeout_secs`) because the HD-PVR video-capture device would start its timestamps at 0 after restarting the source pipeline.

I would hope that any other video-capture device in common use has sensible timestamps, and I don't care about supporting the HD-PVR... but this certainly isn't a pure refactoring.

Reply:
To clarify I don't object to this change but the commit message needs to change and it could possibly be split into several commits and possibly merits a release note.

Reply:
I see.  I hadn't realised that `self.start_timestamp` is reset here: https://github.com/stb-tester/stb-tester/blob/50caf83fa4406970c4690ad0f1f9d887e1db7b45/_stbt/core.py#L1905-L1912

So it can be modified from a separate thread.  Hmm this is racy as well.  Quite a narrow race though.  Hmmmm...

Reply:
This commit does cause the breaking change that now `stbt.frames` can yield 0 frames, whereas before it was guaranteed to yield at least one.  I'll fix this with an explicit check.

Reply:
Good, fixing this explicitly has made CI pass.